src/credential.rs

```diff
@@ -117,6 +131,7 @@ pub trait ProvideAwsCredentials {
 }
 
 /// Provides AWS credentials from environment variables.
+#[derive(Debug, Clone)]
```

Comment:
I see this is used for the threadsafe bits below in this file.  Is there a way we can get around using `derive (Debug, Clone)` ?  Derives are known to increase compile time and we should be sensitive to that. https://github.com/rusoto/rusoto/issues/364 is the parent issue for it.